src/lib/os.pl

```diff
@@ -138,9 +138,8 @@
     can_be(list, Argv),
     '$argv'(Argv0),
     ( member("--", Argv0) ->
-        append(Argv1, ["--"|Argv], Argv0),
-        \+ member("--", Argv1),
-        !
+        once(append(Argv1, ["--"|Argv], Argv0)),
+        \+ member("--", Argv1)
```

Comment:
I think this `\+ member("--", Argv1)` is now entailed and can be removed: `once/1` yields the first solution of the `append/3` goal, and hence `Argv1` cannot contain `"--"`.

Reply:
In fact it seems this can all be written as:

<pre>
(   append(Argv1, ["--"|Argv], Argv0) -> true
;    ...
)
</pre>

or something like it.

Reply:
No, that is not entailed.

e.g. `Argv0 = [ Prefix | "--" , "test1", "--", "test2" | Suffix ]` `Argv = ["test2" | _]`

would result in `Argv1 = [ Prefix | "--" , "test1"]` instead of `Argv1 = Prefix`.

Reply:
> In fact it seems this can all be written as:
> 
> (   append(Argv1, ["--"|Argv], Argv0) -> true
> ;    ...
> )
> or something like it.

Wouldn't that be the original version, which I am trying to fix?

Reply:
Currently running the tests to see if 
```prolog
    can_be(list, Argv),
    '$argv'(Argv0),
    ( append(_, ["--"|Argv1], Argv0) ->
        Argv = Argv1
    ;
        Argv = []
    ).
```

works as expected.

Reply:
Pushed that version as it passed all tests.